File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.py
# [2, 6, 8, 14]
# -> 2 / [1, 3, 4, 7]

def solution(arr):
    maxValue = max(arr)
    count = 1
    
    while True:
        flag = False
        lcm = maxValue * count
        for num in arr:
            if (lcm % num != 0):
                count += 1
                flag = True
                break
        
        if not flag:
            return lcm
                
 

File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.js
function solution(arr) {
    let answer = 0;
    const maxValue = Math.max(...arr);
    const n = arr.length;
    
    let curr = 1;
    while (true) {
        let flag = false;
        const check = maxValue * curr;
        for (let i = 0; i < n; i++) {
            if (check % arr[i] !== 0) {
                flag = true;
                break;
            }
        }
        
        if (flag) {
            curr++;
        } else {
            answer = check;
            break;
        }
    }
    
    return answer;
}